### ISSUE TITLE
Getting NAs while trying to compute Variant effect predictions using basenji_sad.py

### ISSUE THREAD

**Issue by sukritikumar4** [calico/basenji#113]
Hello there,
Thank you for providing access to your analysis and models. Appreciate it deeply.

I hit into a small issue while trying to calculate SNP effects using basenji_sad.py. Here is the process I followed:
-Initially I had a set of 100 variants that I used as inputs, however noticed that I was getting a few NAs while trying to run them. So I decided to use the tutorial data (rs339331) to check if I get these NAs. When I have only one variant I get a SAD score, but If I copy the same variant in a new line (so a duplicate variant in the vcf), I get NAs. Please find the screenshots of the output:

**The VCF file used:**
<img width="373" alt="image" src="https://user-images.githubusercontent.com/23120959/162727418-13e8ed12-53ce-4244-bb0d-50d7181f854f.png">

**The output:**
<img width="919" alt="image" src="https://user-images.githubusercontent.com/23120959/162727518-a8ef60a1-5c97-405c-9635-8f56636ea16f.png">

**The command used:**
_! bin/basenji_sad.py -f data/hg19.ml.fa -o output/ --rc  --shift "1,0,-1" tutorials/models/params_small.json models/heart/model_best.h5 /tutorials/data/rs339331.vcf_

Additionally, I have noticed that the SAD scores change for a variant, if I run it multiple times. Was wondering if my process is correct and why I am hitting into these issues.

**Comment by davek44** [comment]
Hi, these aren't problems that I've seen before. The tutorial models are merely for demonstrations. I'd suggest working with one of our fully trained models, e.g. from here https://github.com/calico/basenji/tree/master/manuscripts/cross2020. If you're still encountering problems, you can email me your VCF, and I'll debug.

**Comment by sukritikumar4** [comment]
Ah, I understand! Thank you so much for your response. 

I just wanted to confirm if what I am doing is right, because my intention is to ultimately use the RF model and get a classification score for a set of non-coding variants with comparative AUCs reported in the publication. Please find the process I have followed so far:
**Step 1** Ran the basenji_sad.py command using params_human.json and model_human.h5 files instead of the old models mentioned in the tutorial.
**Step 2** Received the SAD scores with array dimensions (7207x5313) where the total number of variants I have is 7207
_An observation in Step 2: Of the 7207 variants, there are three variants where the SAD score is NA even with the new model. Perhaps I could email these to you for further debugging._
**Step 3** Use this array as an input for the Human RF classification model to obtain a single classification probability (using predict_proba() for class1).
**Step 4** Use AUC based on Labels and probability.

**Comment by davek44** [comment]
Yes, that workflow looks correct to me. For your remaining NAs, I'd recommend double checking that your VCF and FASTA file given to basenji_sad.py match. Happy to look at it for you, too.

**Comment by sukritikumar4** [comment]
Ah that makes sense, I will check and match the FASTA against the VCF file I have for the three variants. Additionally, I wanted to confirm that the VCF should be aligned with the GrCh37 version and not the GrCh38 version. Would that be correct? (if not, I can imagine that would be the solution to all my issues, I suppose.) 

Thank you so much for your recommendations and Happy Easter weekend! :)

**Comment by davek44** [comment]
You should specify the FASTA for the human genome version you want here: https://github.com/calico/basenji/blob/master/bin/basenji_sad.py#L51
